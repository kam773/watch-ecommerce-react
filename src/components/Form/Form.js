import React from 'react';
import styled from 'styled-components';

export default function Form() {
  return (
      <FormWrapper>
          <form>
              <div className="form-inline">
                  <input class="form-control" id="search" type="text" placeholder="Search" />
                  <button type="submit"><i className="fas fa-search"></i></button>
              </div>
          </form>
      </FormWrapper>
  )
}

const FormWrapper = styled.div`
button {
    background: transparent;
    outline: none;
    border: 1px solid #000;
    padding: 0.4rem 0.6rem;
    cursor: pointer;
}
input {
    border: 1px solid #000;
}

`;