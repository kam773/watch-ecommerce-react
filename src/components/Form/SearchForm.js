import React from "react";
import styled from "styled-components";
import { Form, Input, Button } from "reactstrap";

export default function SearchForm() {
  return (
    <FormWrapper>
      <Form>
        <div className="form-inline">
          <Input
            className="form-control"
            id="search"
            type="text"
            placeholder="Search"
          />
          <Button color="white" type="submit">
            <i className="fas fa-search" />
          </Button>
        </div>
      </Form>
    </FormWrapper>
  );
}

const FormWrapper = styled.div`
  button {
    background: transparent;
    outline: none;
    border: 1px solid #000;
    padding: 0.4rem 0.6rem;
    cursor: pointer;
  }
  input {
    border: 1px solid #000;
  }
`;
