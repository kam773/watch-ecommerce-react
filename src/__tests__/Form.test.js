import React from 'react';
import Form from '../components/Form/Form';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

describe("Form", () => {
    it('should render without crash', () => {
        const tree = renderer.create(<Form/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

})
