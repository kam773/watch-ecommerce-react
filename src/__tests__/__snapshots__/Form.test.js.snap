// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render without crash 1`] = `
<div
  className="sc-bdVaJa cXdaWL"
>
  <form>
    <div
      className="form-inline"
    >
      <input
        className="form-control"
        id="search"
        placeholder="Search"
        type="text"
      />
      <button
        type="submit"
      >
        <i
          className="fas fa-search"
        />
      </button>
    </div>
  </form>
</div>
`;
